pan-domain-auth-play/src/main/scala/com/gu/pandomainauth/action/Actions.scala

```diff
@@ -33,10 +33,10 @@ trait AuthActions {
     */
   def wsClient: WSClient
   def controllerComponents: ControllerComponents
-  def panDomainSettings: PanDomainAuthSettingsRefresher
+  val panDomainSettings: PanDomainAuthSettingsRefresher
 
-  private def system: String = panDomainSettings.system
-  private def domain: String = panDomainSettings.domain
+  private lazy val system: String = panDomainSettings.system
+  private lazy val domain: String = panDomainSettings.domain
   private def settings: PanDomainAuthSettings = panDomainSettings.settings
```

Comment:
Why does this remain a `def`? `settings` seem to be the way to `panDomainSettings` in most instances…

Reply:
Really great question - there are two things that you need to know to understand why we have `def` rather than `val` in `AuthActions` on `def settings: PanDomainAuthSettings = panDomainSettings.settings`.

1. `panDomainSettings` is actually a `PanDomainAuthSettingsRefresher` - it should really be called something like `pandaSettingsRefresher` (tho' changing the name would sadly mean additional disruption for consumers of the Panda library, of course): https://github.com/guardian/pan-domain-authentication/blob/053b9f85c02f907604300503f6ba5d893ea0ccc1/pan-domain-auth-play/src/main/scala/com/gu/pandomainauth/action/Actions.scala#L35

2. `settings` - the field we're reading with `panDomainSettings.settings` - is a field that can **change value over time**, because the `PanDomainAuthSettingsRefresher` will be periodically re-fetching it (and this is essential for key-rotation, as in #150): https://github.com/guardian/pan-domain-authentication/blob/053b9f85c02f907604300503f6ba5d893ea0ccc1/pan-domain-auth-core/src/main/scala/com/gu/pandomainauth/PanDomainAuthSettingsRefresher.scala#L43

...so, given those two things, it **def**initely makes sense that `AuthActions.settings` remains a `def`!